iceoryx2/src/service/builder/publish_subscribe.rs

```diff
@@ -675,12 +677,25 @@ impl<Payload: Debug + ?Sized, UserHeader: Debug, ServiceType: service::Service>
     }
 }
 
+impl<UserHeader: Debug, ServiceType: service::Service> Builder<[u8], UserHeader, ServiceType> {
+    #[doc(hidden)]
+    pub unsafe fn __set_payload_type_details(mut self, value: TypeDetail) -> Self {
```

Comment:
How about `internal_set_...`?

On the other hand, at some point we might need to have this public. In case we have a typed Rust publisher and need to talk with a C subscriber, or vice versa, it might be helpful to be able to control the name of the type on both sides.

Although, then something like this would be more idiomatic

```rs
#[iox2_type_name("bar")]
struct Foo {}
```

Reply:
Why not two `_` but I am open also for `internal`. The most important thing is, that it has to look odd and very weird when someone uses it.

Reply:
To be more explicit. I guess it will show up in code completions and not everybody will understand that `__` is something internal

Reply:
I renamed it into `__internal_***`. As it turned out, I did it all over the place for public functions that are not public. (Forgot my own rules)